orderbook/src/api/validation.rs

```diff
@@ -104,41 +156,117 @@ impl PreOrderValidator {
         }
     }
 
-    pub async fn validate_partial_order(&self, order: PreOrderData) -> Result<(), ValidationError> {
+    pub async fn validate(&self, order: PreOrderData) -> Result<(), PreValidationError> {
         if self.banned_users.contains(&order.owner) {
-            return Err(ValidationError::Forbidden);
+            return Err(PreValidationError::Forbidden);
         }
         if order.buy_token_balance != BuyTokenDestination::Erc20 {
-            return Err(ValidationError::UnsupportedBuyTokenDestination(
+            return Err(PreValidationError::UnsupportedBuyTokenDestination(
                 order.buy_token_balance,
             ));
         }
         if !matches!(
             order.sell_token_balance,
             SellTokenSource::Erc20 | SellTokenSource::External
         ) {
-            return Err(ValidationError::UnsupportedSellTokenSource(
+            return Err(PreValidationError::UnsupportedSellTokenSource(
                 order.sell_token_balance,
             ));
         }
         if order.valid_to
             < shared::time::now_in_epoch_seconds() + self.min_order_validity_period.as_secs() as u32
         {
-            return Err(ValidationError::InsufficientValidTo);
+            return Err(PreValidationError::InsufficientValidTo);
         }
         if has_same_buy_and_sell_token(&order, &self.native_token) {
-            return Err(ValidationError::SameBuyAndSellToken);
+            return Err(PreValidationError::SameBuyAndSellToken);
         }
         if order.buy_token == BUY_ETH_ADDRESS {
             let code_size = self
                 .code_fetcher
                 .code_size(order.receiver)
                 .await
-                .map_err(ValidationError::Other)?;
+                .map_err(PreValidationError::Other)?;
             if code_size != 0 {
-                return Err(ValidationError::TransferEthToContract);
+                return Err(PreValidationError::TransferEthToContract);
+            }
+        }
+        Ok(())
+    }
+}
+
+pub struct PostOrderValidator {
```

Comment:
I feel it is unclear what the distinction between Pre and Post validation is in this file. Would be nice to have a doc comment either on the struct or on the validate function of both components explaining what they check for.

Reply:
Thanks for the great suggestion. Also gave me the idea to combine the two validator structs into one. Also, now it might even make sense to call the methods "validate_partial" and just "validate" since now the validate also calls validate partial.